Updated Extole integration guide to reflect changes on the Extole side (#6953)
braze-inc/braze-docs: _docs/_partners/message_orchestration/channel_extensions/loyalty/extole.md

### Connect to your Braze account

1. Select the Braze integration on the [Partners](https://my.extole.com/partners) page of your My Extole account.
2. Within the Braze integration, hit the Install button to initiate the connection between Extole and Braze.
3. Fill out the required fields, starting with the Braze REST API key. The Braze REST API key can be created in your Braze account and should have the `users.track` option selected. This can be created within your Braze Settings > REST API Key > Create New API Key.
4. Enter your Braze API URL. This URL depends on which instance your Braze account is provisioned to. You can find it [here](https://www.braze.com/docs/api/basics/#endpoints).
5. Add any additional Extole events you'd like to send to Braze beyond the defaults. The default events, event properties, and user attributes are described in the [Extole Events table](https://dev.extole.com/docs/braze#extole-program-events) below.
6. Add any additional Reward states you'd like to send to Braze beyond the default `FULFILLED` state. Refer to the [Extole Rewards table](https://dev.extole.com/docs/braze#extole-rewards) below for a description of all available reward states.

> 7. Select your Braze External ID key mapping, which is how Extole updates user profiles in Braze. You can map the Braze External ID key to Extole's `email_address` or `partner_user_id` for the user.

Reviewer: We might want to add a note here that we recommend against customers using 'email_address' as the 'external_id' as it is not secure. Braze can accept 'email_address' as an identifier but it would be sent within the 'email' field rather than the 'external_id' field.

8. Complete the connection by saving your settings. Once this is done, Extole events will be able to flow into your Braze account.

> ### Extole Program Events

Reviewer 2: ```suggestion
### Extole program events
```

Below are the default events, event properties, and user attributes that are sent into Braze. In addition to the default events listed here, you can add any other Extole events to your integration. Please work with your Extole Implementation or Customer Success Manager to identify and add any additional events you would like to send to Braze.

> | Event | Description | Default Properties & User Attributes |

Reviewer: Within the "Default Properties & User Attributes" which ones are sent as event properties vs. user attributes? Event properties do not consume data points OOTB but user attributes will incur data points so customers will want to be aware of how many they are sending. 

In addition, for those that are user attributes, is this configurable per event? The customer would likely not want to sent "First name" as an attribute with each event since this would incur a data point every time even though the value likely is not changing.

| `extole_shared` | A participant shares their referral link with a friend. | Event name  <br>Event time  <br>Partner (Extole)  <br>Email  <br>External ID  <br>Funnel (advocate or friend)  <br>Program  <br>Share channel  <br>First name  <br>Last name |
| `subscribed` | A participant has opted-in to receive marketing messages. | Email  <br>List type  <br>External ID  <br>Email subscribe (opted in) |
| `unsubscribed` | A participant has opted-out of receiving Extole email communications.| Email  <br>External ID  <br>Subscription state (unsubscribed)  <br>Subscription group ID  <br>List type |
| `outcome` | A participant has converted or completed the desired outcome event configured for the program. | Event name  <br>Event time  <br>Partner (Extole)  <br>Email  <br>External ID  <br>Coupon code  <br>Attributed reward IDs  <br>Partner conversion ID  <br>Value (cart value)  <br>First name  <br>Last name  <br>Share link |
{: .reset-td-br-1 .reset-td-br-2 .reset-td-br-3}

> ### Extole Rewards

Reviewer 2: ```suggestion
### Extole rewards
```

| `FAILED` | An issue has prevented the reward from being issued or sent, requiring attention. | Event name  <br>Event time  <br>Partner  <br>Email  <br>Face value  <br>Coupon code  <br>Face value type  <br>First name  <br>Last name |
| `CANCELED` | The reward has been deactivated and will return to inventory. | Event name  <br>Event time  <br>Partner  <br>Email  <br>Face value  <br>Coupon code  <br>Face value type  <br>First name  <br>Last name |
| `REVOKED` | The fulfilled reward has been invalidated. For example, Extole requested a gift card from a supplier and then subsequently determined that the card was sent in error. If the supplier supports revoking the reward, we would request the funds back and the reward would no longer be valid. | Event name  <br>Event time  <br>Partner  <br>Email  <br>Face value  <br>Coupon code  <br>Face value type  <br>First name  <br>Last name |
{: .reset-td-br-1 .reset-td-br-2 .reset-td-br-3}

## Customization

> ### Lookup and Create Users in Braze

Reviewer 2: ```suggestion
### Lookup and create users in Braze
```

> ### Lookup and create users in Braze

Reviewer 2: ```suggestion
### Find and create users in Braze
```

| Braze REST API key | A Braze REST API key with `users.track` permissions can be created within your Braze Settings > REST API Key > Create New API Key. |
| Braze API URL | Your Braze API URL is specific to your Braze Instance. You can find it [here](https://www.braze.com/docs/api/basics/#endpoints). |
{: .reset-td-br-1 .reset-td-br-2}

## Use cases

The following use cases showcase a few ways you can leverage Extole’s integration with Braze. Work with your Extole implementation and customer success managers to develop an option that fits your company’s specific needs.

- Leverage custom events from your referral and engagement programs to trigger a Braze campaign or Canvas
- Create custom segments, dashboards, and reporting using data from your Extole-powered programs
- Automatically unsubscribe or subscribe users to your marketing list in Braze

## Integration

Complete the following steps to quickly get your integration up and running. Your Extole implementation and customer success managers will support you through this process and answer any questions you may have.

### Connect to your Braze account

> 1. Select the Braze integration on the [Partners](https://my.extole.com/partners) page of your My Extole account.
> 2. Within the Braze integration, hit the Install button to initiate the connection between Extole and Braze.
> 3. Fill out the required fields, starting with the Braze REST API key. The Braze REST API key can be created in your Braze account and should have the `users.track` option selected. This can be created within your Braze Settings > REST API Key > Create New API Key.
> 4. Enter your Braze API URL. This URL depends on which instance your Braze account is provisioned to. You can find it [here](https://www.braze.com/docs/api/basics/#endpoints).
> 5. Add any additional Extole events you'd like to send to Braze beyond the defaults. The default events, event properties, and user attributes are described in the [Extole Events table](https://dev.extole.com/docs/braze#extole-program-events) below.
> 6. Add any additional Reward states you'd like to send to Braze beyond the default `FULFILLED` state. Refer to the [Extole Rewards table](https://dev.extole.com/docs/braze#extole-rewards) below for a description of all available reward states.
> 7. Select your Braze External ID key mapping, which is how Extole updates user profiles in Braze. You can map the Braze External ID key to Extole's `email_address` or `partner_user_id` for the user. Note: We recommend using `external_id` instead of `email_address`as it is more secure.
> 8. Complete the connection by saving your settings. Once this is done, Extole events will be able to flow into your Braze account.

Reviewer 3: ```suggestion
1. Select the Braze integration on the [Partners](https://my.extole.com/partners) page of your My Extole account.
2. In the Braze integration, select **Install** to initiate the connection between Extole and Braze.
3. Fill out the required fields, starting with your Braze REST API key. 
4. Enter your Braze API URL. This URL depends on which instance your Braze account is provisioned to.
5. Add any Extole events you'd like to send to Braze. The default events, event properties, and user attributes are described in the [Extole Events table](https://dev.extole.com/docs/braze#extole-program-events).
6. Add any reward states you'd like to send to Braze other than the `FULFILLED` state. Refer to the [Extole Rewards table](https://dev.extole.com/docs/braze#extole-rewards) for descriptions of the available reward states.
7. Select your Braze external ID key mapping. This is how Extole updates user profiles in Braze. You can map the Braze external ID key to Extole's `email_address` or `partner_user_id` for the user. We recommend using `external_id` instead of `email_address` as it is more secure.
8. Save your settings to complete the connection. Now, Extole events can flow to your Braze account.
```

### Extole program events

> Below are the default events, event properties, and user attributes Extole will send into Braze. In addition to the default events listed here, you can add any other Extole events to your integration. Please work with your Extole Implementation or Customer Success Manager to identify and add any additional events you would like to send to Braze.

Reviewer 3: ```suggestion
Below are the default events, event properties, and user attributes Extole will send to Braze. Reach out to your Extole implementation or customer success managers to identify and add additional Extole events to your integration.
```

| Event | Description | Event Properties | User Attributes |
| ----------- | ----------- | ----------- | ----------- |
| `extole_created_share_link` | A participant creates their share link by entering their email in the Extole Share Experience. | Event name  <br>Event time  <br>Partner (Extole)  <br>Funnel (advocate or friend)  <br>Program | <br>External ID <br>Email  <br>Share link |
| `extole_shared` | A participant shares their referral link with a friend. | Event name  <br>Event time  <br>Partner (Extole)  <br>External ID  <br>Funnel (advocate or friend)  <br>Program  <br>Share channel | Email <br>First name <br>Last Name |

> | `outcome` - The outcome is dynamic based on the configuration of your program (e.g., `extole_shipped`, `extole_converted`, etc.)| A participant has converted or completed the desired outcome event configured for the program. | Dynamic per program | Email <br>First name <br>Last Name |

Reviewer 3: ```suggestion
| `outcome` - The outcome is dynamic based on the configuration of your program (such as `extole_shipped`, `extole_converted`)| A participant has converted or completed the desired outcome event configured for the program. | Dynamic per program | Email <br>First name <br>Last Name |
```

{: .reset-td-br-1 .reset-td-br-2 .reset-td-br-3 .reset-td-br-4}

### Extole subscription states

| Subscription State | Description | Event Properties | User Attributes |
| ----------- | ----------- | ----------- | ----------- |

> | `subscribed` | A participant has opted-in to receive marketing messages. | NA | Email  <br>List type  <br>External ID  <br>Email subscribe (opted in) |

Reviewer 3: ```suggestion
| `subscribed` | A participant has opted-in to receive marketing messages. | N/A | Email  <br>List type  <br>External ID  <br>Email subscribe (opted in) |
```

| `unsubscribed` | A participant has opted-out of receiving Extole email communications.| Email  <br>External ID  <br>Subscription state (unsubscribed)  <br>Subscription group ID  | List type |
{: .reset-td-br-1 .reset-td-br-2 .reset-td-br-3 .reset-td-br-4}

### Extole rewards

> By default, Extole will send reward events in the `FULFILLED` state to Braze so that you can trigger reward notifications via a Braze campaign or canvas. See the table below for additional reward states you may be interested in sending from Extole to Braze.

Reviewer 3: ```suggestion
By default, Extole will send reward events in the `FULFILLED` state to Braze so you can trigger reward notifications via a Braze campaign or Canvas. Refer to the following table for additional reward states.
```

| Reward State | Description | Event Properties | User Attributes |
| ----------- | ----------- | ----------- | ----------- |

> | `FULFILLED` - Default | The reward has been assigned a value (e.g., coupon, gift card, etc.) by an Extole reward supplier. | Email <br>Face value  <br>Coupon code  <br>Face value type  | Email <br>First name  <br>Last name |

Reviewer 3: ```suggestion
| `FULFILLED` | The default state. The reward has been assigned a value (such as coupon or gift card) by an Extole reward supplier. | Email <br>Face value  <br>Coupon code  <br>Face value type  | Email <br>First name  <br>Last name |
```

| `REDEEMED` | The reward has been used by the recipient, as evidenced in a conversion or  redemption event sent to Extole.| Email <br>Face value  <br>Coupon code  <br>Face value type  | Email <br>First name  <br>Last name |
| `FAILED` | An issue has prevented the reward from being issued or sent, requiring attention. | Email <br>Face value  <br>Coupon code  <br>Face value type  | Email <br>First name  <br>Last name |
| `CANCELED` | The reward has been deactivated and will return to inventory. | Email <br>Face value  <br>Face value type  | Email <br>First name  <br>Last name |
| `REVOKED` | The fulfilled reward has been invalidated. For example, Extole requested a gift card from a supplier and then subsequently determined that the card was sent in error. If the supplier supports revoking the reward, we would request the funds back and the reward would no longer be valid. | Email <br>Face value   <br>Face value type  | Email <br>First name  <br>Last name |
{: .reset-td-br-1 .reset-td-br-2 .reset-td-br-3 .reset-td-br-4}

## Customization

### Find and create users in Braze

> For certain use cases, such as a new email or SMS subscription where Extole does not have an external id (user id) for the user, Extole can check for the user's identifier using Braze's Export User by Identifier endpoint. If the user exists within Braze, Extole will add and update any profile attributes. If the request does not return a user profile, Extole will instead use the User Track endpoint to create a User Alias with the user's email address as the Alias Name.

Reviewer 3: ```suggestion
For certain use cases, such as a new email or SMS subscription where Extole does not have an external ID (user ID), Extole can check for the user's identifier using the Braze [Export user profile by identifier endpoint]({{site.baseurl}}/api/endpoints/export/user_data/post_users_identifier/). Extole will add and update any profile attributes if the user exists in Braze. If the request does not return a user profile, Extole will use the `/users/track` endpoint to create a user alias with the user's email address as the alias name.
```
